pilot/cmd/pilot-agent/main.go

```diff
@@ -154,6 +154,27 @@ var (
 			return agent.Run(ctx)
 		},
 	}
+
+	dumpProxyVersionCmd = &cobra.Command{
+		Use:               "version",
+		Short:             "Dump envoy proxy version",
+		PersistentPreRunE: configureLogging,
+		RunE: func(c *cobra.Command, args []string) error {
+			proxy, err := initProxy(args)
+			if err != nil {
+				return err
+			}
+			proxyConfig, err := config.ConstructProxyConfig(meshConfigFile, serviceCluster, options.ProxyConfigEnv, concurrency, proxy)
+			if err != nil {
+				return fmt.Errorf("failed to get proxy config: %v", err)
+			}
+			p := envoy.NewProxy(envoy.ProxyConfig{BinaryPath: proxyConfig.BinaryPath})
```

Comment:
Why not just query admin port /server_info?

Reply:
This command is assumed to be used before running envoy. Currently we have a feature to query `/server_info` but don't have any path to get version before running it.